pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -906,28 +905,20 @@ class Build {
     Executed on a build node, the function checks out the repository and executes the build via ./make-adopt-build-farm.sh
     Once the build completes, it will calculate its version output, commit the first metadata writeout, and archive the build results.
     */
-    def buildScripts(cleanWorkspace, filename) {
+    def buildScripts(cleanWorkspace, cleanWorkspaceAfter, cleanWorkspaceBuildOutputAfter, filename) {
         return context.stage("build") {
 
             if (cleanWorkspace) {
                 try {
 
                     try {
                         context.timeout(time: buildTimeouts.NODE_CLEAN_TIMEOUT, unit: "HOURS") {
-                            if (buildConfig.TARGET_OS == "windows") {
-                                // Windows machines struggle to clean themselves, see:
-                                // https://github.com/AdoptOpenJDK/openjdk-build/issues/1855
-                                context.sh(script: "rm -rf C:/workspace/openjdk-build/workspace/build/src/build/*/jdk/gensrc")
-                                // https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/1419
-                                context.sh(script: "rm -rf J:/jenkins/tmp/workspace/build/src/build/*/jdk/gensrc")
-                                // https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/1662
-                                context.sh(script: "rm -rf E:/jenkins/tmp/workspace/build/src/build/*/jdk/gensrc")
-                                // https://github.com/AdoptOpenJDK/openjdk-infrastructure/issues/1818
-                                context.sh(script: "rm -rf C:/Jenkins/temp/workspace/build/src/build/*/jdk/gensrc")
-                                context.cleanWs notFailBuild: true, disableDeferredWipeout: true, deleteDirs: true
-                            } else {
-                                context.cleanWs notFailBuild: true
-                            }
+                            // Note: Underlying org.apache DirectoryScanner has a bug scanning files containing ".." so must be deleted explicitly
+                            context.println "Cleaning workspace files containing '..'"
+                            context.sh(script: "find . -name \"*..*\"") 
+                            context.sh(script: "find . -name \"*..*\" | xargs rm -f")
```

Comment:
While it seems to work this line scares the ?@#& out of me. I'm nervous about the possibility that it'll find a `..` directory somewhere and point `rm -f` at it ... I suppose it's not recursive though

Reply:
yeah I thought about that too, but as I tested it carefully, and as you said it is not recursive